ext/node/polyfills/process.ts

```diff
@@ -892,52 +857,67 @@ internals.__bootstrapNodeProcess = function (
   core.setMacrotaskCallback(runNextTicks);
   enableNextTick();
 
+  // Install special "unhandledrejection" handler, that will be called
+  // last.
+  internals.nodeProcessUnhandledRejectionCallback = (event) => {
+    if (process.listenerCount("unhandledRejection") === 0) {
+      // The Node.js default behavior is to raise an uncaught exception if
+      // an unhandled rejection occurs and there are no unhandledRejection
+      // listeners.
+      if (process.listenerCount("uncaughtException") === 0) {
+        throw event.reason;
+      }
+
+      event.preventDefault();
+      uncaughtExceptionHandler(event.reason, "unhandledRejection");
+      return;
+    }
+
+    event.preventDefault();
+    process.emit("unhandledRejection", event.reason, event.promise);
+  };
+
+  globalThis.addEventListener("error", (event) => {
+    if (process.listenerCount("uncaughtException") > 0) {
+      event.preventDefault();
+    }
+
+    uncaughtExceptionHandler(event.error, "uncaughtException");
+  });
+
+  globalThis.addEventListener("beforeunload", (e) => {
+    process.emit("beforeExit", process.exitCode || 0);
+    processTicksAndRejections();
+    if (core.eventLoopHasMoreWork()) {
+      e.preventDefault();
+    }
+  });
+
+  globalThis.addEventListener("unload", () => {
+    if (!process._exiting) {
+      process._exiting = true;
+      process.emit("exit", process.exitCode || 0);
+    }
+  });
```

Comment:
It's reverting changes from this commit: https://github.com/denoland/deno/commit/4380a09a0598c73aa434e2f0f3a34555e0bd55cb

Reason for revert: https://github.com/denoland/deno/pull/20892#discussion_r1356879246

Reply:
I understand - but you shouldn't revert changes related to event handling if you are only interested in changes to stdio streams. I'm pretty sure you will get test failures or hangs because of this particular change.

Reply:
oops, fixed